examples/static-file-server/src/main.rs

```diff
@@ -94,9 +94,10 @@ fn using_serve_dir_with_handler_as_service() -> Router {
     }
 
     // you can convert handler function to service
-    let service = handle_404
-        .into_service()
-        .map_err(|err| -> std::io::Error { match err {} });
+    let service = tower::ServiceExt::<Request<Body>>::map_err(
+        handle_404.into_service(),
+        |err| -> std::io::Error { match err {} },
+    );
```

Comment:
Hm having to specific the types like this is pretty unfortunate 😕

Reply:
Feels like between this and the middleware-state thing, tower not receiving nearly as much attention as axum is becoming a problem...